### PR TITLE
ICP-6040 Add fix for updating status after device is plugged in

### PR DESCRIPTION
**Pull request by MMateusiakS**
@greens, when device is connected to power it sends BasicReport with value 0. So refresh() is added. Also we receive 'OK' message, so I ignore it. These changes fix the issue.

### CODE REVIEW

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy [hunk 1]
```diff
@@ -68,13 +68,17 @@ def parse(String description) {
 		result = zwaveEvent(cmd)
 		
 		if(result!=null) {
-			createEvent(result)
+			//device sends 'OK' message when is plugged in
+			if(!result.contains('OK')) {
```
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: where on earth does this "OK" come from?

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy [hunk 2]
```diff
@@ -68,13 +68,17 @@ def parse(String description) {
...
+				createEvent(result)
+			}
 		}
 	}
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {	
 	createEvents(cmd.value)
+	sendHubCommand(refresh())
```
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: you should change the order of these, as you've inadvertently changed the return value

I'm also concerned that by not restricting this to only refresh the children that you have created an infinite loop

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy (hunk 1 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: Yesterday I was in deep concern about this 'OK'. But today I cannot get it anymore.Let's assume that I made huge mistake.

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy (hunk 2 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: Ok. I change the order. Also I created method for refreshing only children.

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy [hunk 3]
```diff
@@ -74,6 +74,9 @@ def parse(String description) {
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {	
+	if(cmd.value == 0) {
```
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy:80]: Can you add a comment that explains why this is done.

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy [hunk 4]
```diff
@@ -144,6 +147,14 @@ def refresh() {
 	return delayBetween(cmds, 200)
 }
 
+def refreshChildren() {
```
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy:150]: use this method to remove the duplicated code in `refresh()`

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy (hunk 3 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy:80]: comment added

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy (hunk 4 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy:150]: done

**Comment by tpmanley** [comment]
I was asked to get this into staging this week so opened up https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3534 and will close this.

**Pull request closed without merging**